zerver/lib/bugdown/__init__.py

```diff
@@ -1904,22 +1907,19 @@ def get_possible_mentions_info(realm_id: int, full_names: Set[str]) -> Dict[str,
         'full_name',
         'email',
     )
-    dct = {}  # type: Dict[str, FullNameInfo]
-    for row in rows:
-        key = row['full_name'].lower()
-        # To insert users with duplicate full names in the dict
-        if key in dct:
-            key = '{}|{}'.format(key, row['id'])
-        dct[key] = row
-    return dct
+    return list(rows)
 
 class MentionData:
     def __init__(self, realm_id: int, content: str) -> None:
-        full_names = possible_mentions(content)
-        self.full_name_info = get_possible_mentions_info(realm_id, full_names)
+        mention_texts = possible_mentions(content)
```

Comment:
This part needed to be changed to use the list instead of the dict. Essentially, we generate two dicts here, and both are now generated slightly differently with the end result being the same.